src/core/network/qgsnetworkaccessmanager.cpp

```diff
@@ -115,6 +115,59 @@ class QgsNetworkProxyFactory : public QNetworkProxyFactory
 };
 ///@endcond
 
+
+/// @cond PRIVATE
+class QgsNetworkCookieJar : public QNetworkCookieJar
+{
+  public:
```

Comment:
```suggestion
class QgsNetworkCookieJar : public QNetworkCookieJar
{
  Q_OBJECT
  public:
```

Reply:
This would  result in `contains a "Q_OBJECT" macro, but does not include "qgsnetworkaccessmanager.moc"!` Think one can do without Q_OBJECT here?

Reply:
Just move the class declaration to the header and it should work ok

Reply:
Yes -- qt docs clearly state that the macro is needed for ALL qobject derived classes

Reply:
From the docs: "We strongly recommend the use of this macro in all subclasses of QObject regardless of whether or not they actually use signals, slots and properties, since failure to do so may lead certain functions to exhibit strange behavior."

Reply:
Isn't that solved by adding `#include "qgsnetworkaccessmanager.moc"` at the bottom of this files (like we do for tests)

Reply:
Done